src/Tickets/Commerce/Flag_Actions/Decrease_Stock.php

```diff
@@ -105,7 +105,18 @@ public function handle( Status_Interface $new_status, $old_status, \WP_Post $pos
 				continue;
 			}
 
-			update_post_meta( $ticket->ID, Ticket::$stock_meta_key, $ticket->stock() - $quantity );
+			/**
+			 * Filter the stock update value for a ticket.
+			 *
+			 * @since TBD
+			 *
+			 * @param int      $update_stock The updated stock value.
+			 * @param Ticket   $ticket       The ticket object.
+			 * @param \WP_Post $post         The order post object.
+			 */
+			$update_stock = apply_filters( 'tec_tickets_commerce_decrease_ticket_stock', $ticket->stock() - $quantity, $ticket, $post );
+
+			update_post_meta( $ticket->ID, Ticket::$stock_meta_key, $update_stock );
```

Comment:
This method, as well as the other one in the `Increase_Stock` action, are very specific to Tickets Commerce.
I would suggest a more general-purpose method, still based around the sync of the `_stock` meta key (which is the same for all providers) and "sitting" on the updates of the `_stock` meta key.

The logic would not change: when an update to the `_stock` meta key comes in for an ASC ticket, set the stock of all other ASC tickets sharing the seat type (the one dictating the initial `_stock` value) to the new value.
Remove the filtering method temporarily to avoid recursions while running `update_post_meta( $id, '_stock', $value )` on the other tickets, and keep invalidating the cache like you're doing now.